src/lib/alloc.c

```diff
@@ -191,9 +191,41 @@ static void *rmalloc_sys(int zone, int caps, int core, size_t bytes)
 	return ptr;
 }
 
+static void *align_buffer_ptr(struct mm_heap *heap, uint32_t alignment,
+			      void *ptr)
+{
+	struct alignment_hdr *a_hdr;
+	void *aligned_ptr = ptr;
+	int i, mod_align;
+
+	/* Checking alignment for buffer heap only */
+	for (i = 0;  i < PLATFORM_HEAP_BUFFER;  i++) {
+		if (heap == &memmap.buffer[i]) {
```

Comment:
If content of the for body is just this if you can change it to early continue instead of nesting whole for logic.